### ISSUE TITLE
failed to retrace a package from backports (obtained via local debproxy)

### ISSUE THREAD

**Issue by yarikoptic** [ReproNim/niceman#132]
```shell
$> apt-cache policy python-six                              
python-six:
  Installed: 1.10.0-3~bpo8+1~nd80+1
  Candidate: 1.10.0-3~bpo8+1~nd80+1
  Version table:
     1.10.0-3~bpo8+1 0
        100 http://debproxy:9999/debian/ jessie-backports/main amd64 Packages
 *** 1.10.0-3~bpo8+1~nd80+1 0
        500 http://neuro.debian.net/debian/ jessie/main amd64 Packages
        100 /var/lib/dpkg/status
     1.8.0-1 0
        500 http://debproxy:9999/debian/ jessie/main amd64 Packages

$> apt-cache policy | grep -3 backports
 500 http://neuro.debian.net/debian-devel/ jessie/main amd64 Packages
     release o=NeuroDebian,a=jessie,n=jessie,l=NeuroDebian,c=main
     origin neuro.debian.net
 500 http://debproxy:9999/debian/ jessie-backports/non-free Translation-en
 500 http://debproxy:9999/debian/ jessie-backports/main Translation-en
 500 http://debproxy:9999/debian/ jessie-backports/contrib Translation-en
 100 http://debproxy:9999/debian/ jessie-backports/non-free amd64 Packages
     release o=Debian Backports,a=jessie-backports,n=jessie-backports,l=Debian Backports,c=non-free
     origin debproxy
 100 http://debproxy:9999/debian/ jessie-backports/contrib amd64 Packages
     release o=Debian Backports,a=jessie-backports,n=jessie-backports,l=Debian Backports,c=contrib
     origin debproxy
 100 http://debproxy:9999/debian/ jessie-backports/main amd64 Packages
     release o=Debian Backports,a=jessie-backports,n=jessie-backports,l=Debian Backports,c=main
     origin debproxy
 500 http://neuro.debian.net/debian-local/ jessie/non-free amd64 Packages
     release o=NeuroDebianLocal,n=jessie,l=NeuroDebianLocal,c=non-free
(dev)1 10017.....................................:Thu 02 Nov 2017 02:36:40 PM EDT:.
(git)head1:~/proj/repronim/niceman[master]
$> niceman retrace /usr/lib/python2.7/dist-packages/six.py      
/home/yoh/proj/repronim/niceman/venvs/dev/local/lib/python2.7/site-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.22) or chardet (2.3.0) doesn't match a supported version!
  RequestsDependencyWarning)
2017-11-02 14:37:18,194 [INFO   ] Tracing using <class 'niceman.distributions.debian.DebTracer'> 
2017-11-02 14:37:18,702 [ERROR  ] expected string or buffer [debian.py:parse_apt_cache_policy_source_info:218] (TypeError) 
(dev)1 10018 ->1.....................................:Thu 02 Nov 2017 02:37:18 PM EDT:.
(git)head1:~/proj/repronim/niceman[master]
$> niceman --dbg retrace /usr/lib/python2.7/dist-packages/six.py
/home/yoh/proj/repronim/niceman/venvs/dev/local/lib/python2.7/site-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.22) or chardet (2.3.0) doesn't match a supported version!
  RequestsDependencyWarning)
2017-11-02 14:37:26,138 [INFO   ] Tracing using <class 'niceman.distributions.debian.DebTracer'> 
Traceback (most recent call last):
  File "/home/yoh/proj/repronim/niceman/venvs/dev/bin/niceman", line 9, in <module>
    load_entry_point('niceman==0.0.2', 'console_scripts', 'niceman')()
  File "/home/yoh/proj/repronim/niceman/niceman/cmdline/main.py", line 238, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/home/yoh/proj/repronim/niceman/niceman/interface/base.py", line 352, in call_from_parser
    return cls.__call__(**kwargs)
  File "/home/yoh/proj/repronim/niceman/niceman/interface/retrace.py", line 95, in __call__
    session=session
  File "/home/yoh/proj/repronim/niceman/niceman/interface/retrace.py", line 161, in identify_distributions
    files_to_trace):
  File "/home/yoh/proj/repronim/niceman/niceman/distributions/debian.py", line 242, in identify_distributions
    packages, remaining_files = self.identify_packages_from_files(files)
  File "/home/yoh/proj/repronim/niceman/niceman/distributions/base.py", line 230, in identify_packages_from_files
    pkg = self._create_package(**pkgfields)
  File "/home/yoh/proj/repronim/niceman/niceman/distributions/debian.py", line 308, in _create_package
    self._find_all_sources()
  File "/home/yoh/proj/repronim/niceman/niceman/distributions/debian.py", line 353, in _find_all_sources
    src_info = parse_apt_cache_policy_source_info(out)
  File "/home/yoh/proj/repronim/niceman/niceman/support/distributions/debian.py", line 218, in parse_apt_cache_policy_source_info
    attribs = re_rel_attrib.finditer(info.get("release_info"))
TypeError: expected string or buffer
()
> /home/yoh/proj/repronim/niceman/niceman/support/distributions/debian.py(218)parse_apt_cache_policy_source_info()
-> attribs = re_rel_attrib.finditer(info.get("release_info"))
(Pdb) p info
{'priority': u'500', 'source': u'http://debproxy:9999/debian/ jessie-backports/non-free Translation-en', 'origin_info': None, 'release_info': None}
```

**Comment by rbuccigrossi** [comment]
Aha!  Actually the issue is not with the parsing of the lines themselves, but that in your environment you have:

 500 http://debproxy:9999/debian/ jessie-backports/contrib Translation-en
 100 http://debproxy:9999/debian/ jessie-backports/non-free amd64 Packages
     release o=Debian Backports,a=jessie-backports,n=jessie-backports,l=Debian Backports,c=non-free
     origin debproxy

My code expected a "release" line for every entry. Why doesn't jessie-backports/contrib have a "release line?  However, adding a judicious test for an empty release_info fixes that.

**Comment by rbuccigrossi** [comment]
This is addressed in f48e79f

**Comment by yarikoptic** [comment]
yeap, seems to work, thanks!
